modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesKernelBoot.java

```diff
@@ -335,10 +343,35 @@ public static void startHouseKeepingThreads() throws AndesException {
 
         //reload exchanges/queues/bindings and subscriptions
         AndesRecoveryTask andesRecoveryTask = new AndesRecoveryTask();
-        Integer scheduledPeriod = AndesConfigurationManager.readValue
+        //deleted the expired message from db
+        PeriodicExpiryMessageDeletionTask periodicExpiryMessageDeletionTask = null;
+
+        int recoveryTaskScheduledPeriod = AndesConfigurationManager.readValue
                 (AndesConfiguration.PERFORMANCE_TUNING_FAILOVER_VHOST_SYNC_TASK_INTERVAL);
-        andesRecoveryTaskScheduler.scheduleAtFixedRate(andesRecoveryTask, scheduledPeriod,
-                                                       scheduledPeriod, TimeUnit.SECONDS);
+        int dbBasedDeletionTaskScheduledPeriod = AndesConfigurationManager.readValue
+                (AndesConfiguration.PERFORMANCE_TUNING_PERIODIC_EXPIRY_MESSAGE_DELETION_INTERVAL);
+        int safeDeleteRegionSlotCount = AndesConfigurationManager.readValue
+                (AndesConfiguration.PERFORMANCE_TUNING_SAFE_DELETE_REGION_SLOT_COUNT);
+        boolean isDLCExpiryCheckEnabled = AndesConfigurationManager.readValue
+                (AndesConfiguration.PERFORMANCE_TUNING_EXPIRE_MESSAGES_IN_DLC);
+
+        //based on the DLC expiration check configuration bind the appropriate deletion task
+        if (isDLCExpiryCheckEnabled) {
```

Comment:
Are 'isExpiryCheckEnabledInDLC' and 'isDLCExpiryCheckEnabled'? We have duplicated the variable, right?